lib/ipam/ipam.go

```diff
@@ -712,6 +762,7 @@ func (c ipamClient) autoAssign(ctx context.Context, num int, handleID *string, a
 						break
 					}
 					logCtx.Infof("Successfully assigned IPs from non-affine block %s", blockCIDR.String())
+					iPAMAssignmentInfo.NumAssigned += len(newIPs)
```

Comment:
Rather than try to catch all the places we might allocate IPs, could we set NumAssigned at the end of the function?

Or going a step further, see my suggestion about just embedding the allocated IPs in the struct rather than having two separate return values. WDYT?

Reply:
Yes, that was @mgleung's suggestion as well (embedding the `[]cnet.IPNet` slice inside the struct), and again I agree it's better! Will do that.